saltproc/tests/integration_tests/no_repro/test_no_reproc_run.py

```diff
@@ -36,18 +36,65 @@
                         sim_depcode=depcode,
                         core_number=1,
                         node_number=1,
-                        h5_file=db_file,
-                        iter_matfile=iter_matfile)
+                        db_path=db_file)
 
 tap = Reactor(volume=1.0,
               power_levels=[1.250E+09],
               depl_hist=[3])
 
 
+def runsim_no_reproc(simulation, reactor, nsteps):
+    """Run simulation sequence for integration test. No reprocessing
+    involved, just re-running depletion code for comparision with model
+    output.
+
+    Parameters
+    ----------
+    reactor : `Reactor`
+        Contains information about power load curve and cumulative
+        depletion time for the integration test.
+    nsteps : int
+        Number of depletion time steps in integration test run.
+
+    """
+
+    ######################################################################
+    # Start sequence
+    for dep_step in range(nsteps):
+        print("\nStep #%i has been started" % (dep_step + 1))
+        if dep_step == 0:  # First step
+            simulation.sim_depcode.write_depcode_input(
+                reactor,
+                dep_step,
+                False)
+            simulation.sim_depcode.run_depcode(
+                simulation.core_number,
+                simulation.node_number)
+            # Read general simulation data which never changes
+            simulation.store_run_init_info()
+            # Parse and store data for initial state (beginning of dep_step)
+            mats = simulation.sim_depcode.read_dep_comp(
+                False)
+            simulation.store_mat_data(mats, dep_step, False)
+        # Finish of First step
+        # Main sequence
+        else:
+            simulation.sim_depcode.run_depcode(
+                simulation.core_number,
+                simulation.node_number)
+        mats = simulation.sim_depcode.read_dep_comp(
+            True)
+        simulation.store_mat_data(mats, dep_step, False)
+        simulation.store_run_step_info()
+        simulation.sim_depcode.write_mat_file(
+            mats,
+            simulation.burn_time)
+
+
 @pytest.mark.slow
 # @pytest.mark.skip
 def test_integration_3step_saltproc_no_reproc_heavy():
-    simulation.runsim_no_reproc(tap, 2)
+    runsim_no_reproc(simulation, tap, 2)
```

Comment:
Yeah, so this should probably be a 3 just to reduce confusion, but the reason it is a 2 has to do with the function `runsim_no_reproc`. The first line in that function is this:
```
for dep_step in range(nsteps):
```
Because python is 0-indexed, `range(2)` will give a list `[0, 1, 2]`, which is 3 steps.

This isn't descriped in the docstring for `runsim_no_reproc`, so I'll decrement the variable in the function by 1 and increase the input by 1.

Reply:
Wait, doesn't the range function, when given one argument - aka range(n) - give a list of integers starting at zero, *length* n?  As in, range(5) is 0, 1, 2, 3, 4.  source: https://cs.stanford.edu/people/nick/py/python-range.html

Reply:
@ZoeRichter You are absolutely right and it seems I was confused. Perhaps the test name is a typo and it really is supposed to be a 2-step?

Reply:
I reverted ad86976 and ran the `no_repro` integration test with 3 steps (instead of the 2 it normally has) and it failed. Looking at the data file used for comparing the data of the saltproc simulation shows that there are indeed data for three depletion steps:
```
MAT_fuel_ADENS = [
0.00000E+00 5.86157E-10 1.17631E-09 1.76756E-09 % H1
0.00000E+00 2.97625E-11 5.96570E-11 8.96147E-11 % H2
0.00000E+00 2.02909E-09 4.05054E-09 6.06415E-09 % H3
``` 
(the first column is the initial fuel composition). 

Interestingly, the integration test only does 2 depletion steps when running saltproc, and this is also what the error file (`sss_vs_sp_no_reproc_error`) is based on. I suggest we rename the integration test to `...2step...` as it is more accurate, even though data appears to exist for up to three depletion steps.

Reply:
Is only running 2 depletion steps an expected behavior for the integration test?  Is it supposed to fail for more than 2 steps?

Reply:
Answer to your first question is yes. Answer to your second question is that the integration test is checking a precaclulated result with the result from the simulation. The pre-calculated result matches what the simulations' solution is for two depletion steps. It fails due to an assertion error:
```
E       AssertionError: 
E       Arrays are not equal
E       
E       Mismatched elements: 1489 / 1629 (91.4%)
E       Max absolute difference: 0.000212
E       Max relative difference: 147991.49999825
E        x: array([-1.62895e-09, -2.55284e-10, -1.00741e-08, ...,  0.00000e+00,
E               0.00000e+00,  0.00000e+00])
E        y: array([-2.6046e-10, -6.1869e-11, -7.2800e-11, ...,  0.0000e+00,
E               0.0000e+00,  0.0000e+00])

saltproc/tests/integration_tests/no_repro/test_no_reproc_run.py:107: AssertionError
----------------------------- C
```